### ISSUE TITLE
Question / Suggestion: Consider using elm-hot?

### ISSUE THREAD

**Issue by mdevlamynck** [wking-io/elm-live#146]
[klazuka/elm-hot](https://github.com/klazuka/elm-hot) provides the low-level features for hot swapping elm code with model state preservation between reloads (only 0.19).

Would it make sense for this project to use it?

**Comment by wking-io** [comment]
I have actually discussed with klazula doing just that, but haven't gotten around to it. If you are interested in taking a stab at it feel free!

**Comment by lucamug** [comment]
Ok, I have done some experimenting about hot swap and I was able to make it working in elm-live.

Few caveats:

The script injected by elm-live need to run in the browser BEFORE the elm compiled javascript. So I moved it before everything. I am afraid this is going to break the html so we probably need to be smarter here (adding it just after `<head>`?)

We need to extrapolate the relative path of the elm compiled javascript. For the moment I used one argument of `elm make` (`--output`) and one argument of `elm-live` (`dir`). This should probably be done in a more robust way

Probably we need a configuration flag to toggle this feature (simple)

The code for testing is available at https://github.com/lucamug/elm-live/tree/hot-swap

I will start using this version at work to see if it is reliable

![elm-live-hot-swap](https://user-images.githubusercontent.com/5551094/58694985-3ad4bc00-83cf-11e9-87b3-2024532112c6.gif)
